base/database/batch.go

```diff
@@ -0,0 +1,233 @@
+package database
+
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/jinzhu/gorm"
+)
+
+// bulkNow holds a global now state and will be used for each records as
```

Comment:
Is this code copied from anywhere? We should reference it.

Reply:
Fixed.